pkg/apis/eventing/v1/trigger_validation.go

```diff
@@ -163,3 +149,74 @@ func (t *Trigger) validateInjectionAnnotation(injectionAnnotation string) *apis.
 	}
 	return nil
 }
+
+func ValidateAttributeFilters(filter *TriggerFilter) (errs *apis.FieldError) {
+	if filter == nil {
+		return nil
+	}
+	return errs.Also(ValidateAttributesNames(filter.Attributes).ViaField("attributes"))
+}
+
+func ValidateAttributesNames(attrs map[string]string) (errs *apis.FieldError) {
+	for attr := range attrs {
+		if !validAttributeName.MatchString(attr) {
+			errs = errs.Also(apis.ErrInvalidKeyName(attr, apis.CurrentField, "Attribute name must start with a letter and can only contain lowercase alphanumeric").ViaKey(attr))
+		}
+	}
+	return errs
+}
+
+func ValidateSingleAttributeMap(expr map[string]string) (errs *apis.FieldError) {
+	if len(expr) == 0 {
+		return nil
+	}
+
+	if len(expr) != 1 {
+		return apis.ErrGeneric("Multiple items found, can have only one key-value", apis.CurrentField)
+	}
+	for attr := range expr {
+		if !validAttributeName.MatchString(attr) {
+			errs = errs.Also(apis.ErrInvalidKeyName(attr, apis.CurrentField, "Attribute name must start with a letter and can only contain lowercase alphanumeric").ViaKey(attr))
+		}
+	}
+	return errs
+}
+
+func ValidateSubscriptionAPIFiltersList(ctx context.Context, filters []SubscriptionsAPIFilter) (errs *apis.FieldError) {
+	if filters == nil || !feature.FromContext(ctx).IsEnabled(feature.NewTriggerFilters) {
+		return nil
+	}
+
+	for i, f := range filters {
+		f := f
+		errs = errs.Also(ValidateSubscriptionAPIFilter(ctx, &f)).ViaIndex(i)
+	}
+	return errs
+}
+
+func ValidateSubscriptionAPIFilter(ctx context.Context, filter *SubscriptionsAPIFilter) (errs *apis.FieldError) {
+	if filter == nil {
+		return nil
+	}
+	errs = errs.Also(
+		ValidateOneOf(filter),
+	).Also(
+		ValidateSingleAttributeMap(filter.Exact).ViaField("exact"),
+	).Also(
+		ValidateSingleAttributeMap(filter.Prefix).ViaField("prefix"),
+	).Also(
+		ValidateSingleAttributeMap(filter.Suffix).ViaField("suffix"),
+	).Also(
+		ValidateSubscriptionAPIFiltersList(ctx, filter.All).ViaField("all"),
+	).Also(
+		ValidateSubscriptionAPIFiltersList(ctx, filter.Any).ViaField("any"),
+	).Also(ValidateSubscriptionAPIFilter(ctx, filter.Not).ViaField("not"))
```

Comment:
Nit: for consistent formatting:
```suggestion
		ValidateSubscriptionAPIFiltersList(ctx, filter.Any).ViaField("any"),
	).Also(
	        ValidateSubscriptionAPIFilter(ctx, filter.Not).ViaField("not"),
	)
```